tests/test_relations.py

```diff
@@ -48,6 +64,35 @@ def test_pk_representation(self):
         assert representation == self.instance.pk
 
 
+class TestHyperlinkedRelatedField(APISimpleTestCase):
+    urls = 'tests.test_relations'
+
+    def setUp(self):
+        class HyperlinkedMockQueryset(MockQueryset):
+            def get(self, **lookup):
+                for item in self.items:
+                    if item.pk == int(lookup.get('pk', -1)):
+                        return item
+                raise ObjectDoesNotExist()
+
+        self.queryset = HyperlinkedMockQueryset([
+            MockObject(pk=1, name='foo'),
+            MockObject(pk=2, name='bar'),
+            MockObject(pk=3, name='baz')
+        ])
+        self.field = serializers.HyperlinkedRelatedField(
+            view_name='example-detail',
+            queryset=self.queryset
+        )
+        request = factory.post('/')
+        request.versioning_scheme = NamespaceVersioning()
+        self.field._context = {'request': request}
+
+    def test_bug_2489(self):
+        self.field.to_internal_value('/example/3/')
+        self.field.to_internal_value('/v1/example/3/')
+
+
```

Comment:
How about if we put this in `tests/versioning.py`?
I'd also prefer if we can keep the urlpatterns isolated to the same module as the test.